src/main/java/bruno/task/TaskList.java

```diff
@@ -0,0 +1,197 @@
+package bruno.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+
+import bruno.Bruno;
+import bruno.Storage;
+import bruno.exceptions.BrunoException;
+import bruno.exceptions.EmptyTaskException;
+import bruno.exceptions.InvalidTaskIndexException;
+import bruno.exceptions.MissingFieldException;
+import bruno.exceptions.UnknownCommandException;
+
+/**
+ * Manages the list of tasks, including adding, marking, unmarking, and deleting tasks.
+ * Interacts with the Storage class to load and save tasks from and to a file.
+ */
+public class TaskList {
+    private DateTimeFormatter formatter1 = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+    private ArrayList<Task> tasks;
```

Comment:
Perhaps you could standardise the naming of `tasks`? There was a task list earlier that was named `taskList`.